### ISSUE TITLE
Width preference does not save

### ISSUE THREAD

**Issue by jarsudsco** [grosenberg/IndentGuide#7]
I have IndentGuide 1.6.0.20191128-0105 running in Eclipse:

Eclipse IDE for C/C++ Developers (includes Incubating components)
Version: 2020-06 (4.16.0)
Build id: 20200615-1200
OS: Windows 10, v.10.0, x86_64 / win32
Java version: 13.0.1

The width preference does not appear to save.  Each time I restart Eclipse width resets to 4 which is way too wide for my liking.  All other preferences - alpha, style, shift, colour and the drawing options, appear to correctly save and reload on startup.

**Comment by grosenberg** [comment]
Works correctly for me. The builtin default line width is '1'.  

Do you have Oomph enabled?  Might be what is (re)setting the value to '4'.

**Comment by jarsudsco** [comment]
Hi thanks for the reply.  Sorry I'm not an Eclipse expert - I know of Oomph, I thought it was an integral part of Eclipse and I did not realise I had any control over it - so I have not changed any Oomph settings that I am aware of.  Seem a rather specific thing to reset, but how can I check that?  AFAIK 4 was the width value from the moment I installed IntentGuide - I would never have chosen that - I always use 1 (as I did in previous incarnations of IndentGuide)

**Comment by grosenberg** [comment]
It is used to sync (enforce) preference settings across teams - not generally something for solo devs. 

Go to `Preferences -> Oomph -> Setup Tasks` and uncheck everything & choose `Never`. On the subpages, uncheck everything. Apply/Close.

Cannot hurt to turn it off. Of course, if Oomph was not the source of the problem, nothing will have changed.

**Comment by jarsudsco** [comment]
Thanks for the suggestion...

None of the Oomph options were checked anyway but I did set the startup option to "never" as you suggested.  Restarted Eclipse.  Set the IndentGuide width to 1, applied it, closed Eclipse, restarted, width was back to 4...

**Comment by grosenberg** [comment]
Maybe the IndentGuide instance is somehow corrupt. Try uninstall/reinstall.

Maybe some other plugin is interfering.  Try disabling everything other than the platform.

Beyond that, I am all out of ideas.

**Comment by jarsudsco** [comment]
Thanks again for the suggestions, I did the following:

Uninstalled IndentGuide
Restarted Eclipse with -clean option then exited
Ran the Eclipse installer and cleaned up the bundle pools
Restarted Eclipse
Re-installed IndentGuide
Checked preferences and the line width was 4

It seems to me there are 3 possibilities:

1. The line width is not being saved correctly
2. The saved line width is not being reinstated correctly
3. Something else is changing it as you suggested

Presumably, I thought, the preferences would be stored in a file so I searched for filenames containing the string "indentguide".

I found a filed named net.certiv.tools.indentguide.prefs in a folder under my Eclipse workspace here:

    **<workspace>\.metadata\.plugins\org.eclipse.core.runtime\.settings\net.certiv.tools.indentguide.prefs**

(I'm surprised this wasn't mentioned straight away.)

This file contained the following:

```
    eclipse.preferences.version=1
    indentguide.content_types={long value omitted for clarity}
    indentguide.line_alpha=64
    indentguide.line_color=128,128,128
    indentguide.line_shift=3
    indentguide.line_style=3
```

So no reference to line width, but this was when Eclipse was not running.  By chance, I ran Eclipse and my file editor announced that the file it was editing had changed - did I want to reload it...  I said yes and the following 2 lines had been added to the prefs file:

```
    indentguide.line_width=4
    overriddenByCSS=,indentguide.line_width,
```

If I close Eclipse, these 2 lines are removed again.  Not great to have a *preferences* file that changes to reflect some sort of *status*.

Taking the CSS reference as a hint, I then searched for CSS files that contained the string "indentguide".

I found a file named color-definitions-fix.css here:

    **<workspace>\.metadata\.plugins\com.genuitec.eclipse.ui.common.platform\color-definitions-fix.css**

and within that file I found the following:

```
    IEclipsePreferences#net-certiv-tools-indentguide {
        preferences:
            'indentguide.line_color=185,192,198'
            'indentguide.line_width=4'
    }

```
Why line width would be in a color definitions file anyway I don't know, but I guess clearly that's where the annoying "4" was coming from.

I removed the entire "IEclipsePreferences#net-certiv-tools-indentguide" section from the css file, added just the line "indentguide.line_width=1" to the prefs file restarted Eclipse and the line width was correctly set to 1 in the IndentGuide preferences page.

More curiously... if I changed the line width on the IndentGuide preferences page and applied it - the prefs file would be correctly updated to reflect the new line width value for every value EXCEPT 1.  If I choose a line width of 1, the "indentguide.line_width" line disappears from the prefs file.

The "IEclipsePreferences#net-certiv-tools-indentguide" section that I deleted from the css file appears to get re-created every time the IndentGuide preferences page is used, but re-created with exactly the values shown above - a color I haven't chosen and a line width of 4 - not whatever is actually on the preferences page.

Presumably because there is then no "indentguide.line_width" line in the prefs file, on subsequent startups, the incorrect line width "preference" from the css file is copied to the real preferences file with the additional "overiddenByCSS" line (which kind of looks mal-formed), and so I'm stuck at a line width of 4 again. 

So any value of line width from 2-8 set via the IndentGuide preferences page will be correctly saved and recovered, but the value I want, 1, causes quite strange and presumably not by design behaviour.

Currently, I have a workaround - I can manually add the line "indentguide.line_width=1" to the prefs file and this persists, unless I change the line width value to something other than 1 then back to 1 on the preferences page (most unlikely).  The other preferences don't seem to affect this so I can live with this, however something is clearly broken.

I think the questions that remain are:

1. Why is a line width value of 1 causing a failure to write the corresponding line into the preferences file?

and

2. Why is the "IEclipsePreferences#net-certiv-tools-indentguide" section being re-created in the css file with values unrelated to the actual preferences?
